src/SFA.DAS.ApprenticeCommitments.Web/Pages/Overview.cshtml.cs

```diff
@@ -1,11 +1,35 @@
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
 using Microsoft.AspNetCore.Mvc.RazorPages;
+using SFA.DAS.ApprenticeCommitments.Web.Services.OuterApi;
+using SFA.DAS.HashingService;
+using System.Threading.Tasks;
 
 namespace SFA.DAS.ApprenticeCommitments.Web.Pages
 {
+    [Authorize]
     public class OverviewModel : PageModel
     {
-        public void OnGet()
+        private readonly IOuterApiClient _client;
+        private readonly IHashingService _hashing;
+
+        public OverviewModel(IOuterApiClient client, IHashingService hashing)
+        {
+            _client = client;
+            _hashing = hashing;
+        }
+
+        public async Task<IActionResult> OnGet([FromServices] AuthenticatedUser user)
+        {
+            return await RedirectToLatestApprenticeship(user);
+        }
+
+        private async Task<IActionResult> RedirectToLatestApprenticeship(AuthenticatedUser user)
         {
+            var apprenticeship = await _client.GetApprenticeships(user.RegistrationId);
+            var firstApprenticeship = apprenticeship[0];
+            var hashedId = _hashing.HashValue(firstApprenticeship.Id);
+            return RedirectToPage("Apprenticeships", new { apprenticeshipId = hashedId });
```

Comment:
this just into an infinite redirect look when running locally with MockApi

Reply:
This was caused by a bug in the page model binding